google/cloud/pubsub/publisher_options.cc

```diff
@@ -20,6 +20,10 @@ namespace cloud {
 namespace pubsub {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
+PublisherOptions::PublisherOptions() {
```

Comment:
nit: we could save these lines by only offering a `Publisher(Options opts = {})`. This is fine though.

Reply:
Will keep as-is, mostly laziness.